### PR TITLE
(refactor) O3-4251 : Update dispensing app to use workspaces for  launch forms

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
see https://openmrs.atlassian.net/browse/O3-4251

## Screenshots
[Kapture 2024-12-04 at 22.20.18.webm](https://github.com/user-attachments/assets/fa38f010-6961-4359-ba7a-1bf4daad2037)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -33,3 +36,8 @@ export const dispensingDashboardLink = getSyncLifecycle(dispensingLinkHomepageCo
 export const dispenseActionButton = getSyncLifecycle(DispenseActionButton, options);
 export const pauseActionButton = getSyncLifecycle(PauseActionButton, options);
 export const closeActionButton = getSyncLifecycle(CloseActionButton, options);
+
+// Dispensing workspace
+export const dispenseWorkspace = getSyncLifecycle(DispenseForm, options);
+export const closeDispenseWorkspace = getSyncLifecycle(CloseDispenseForm, options);
+export const pauseDispenseWorkspace = getSyncLifecycle(PauseDispenseForm, options);
```
Comment (ibacher) [inline src/index.ts:43]: For workspaces, we should always use the `getAsyncLifecycle()` pattern

File: src/history/history-and-comments.component.tsx [hunk 2]
```diff
@@ -226,7 +232,7 @@ const HistoryAndComments: React.FC<{
                   {dispense.performer && dispense.performer[0]?.actor?.display} {generateDispenseVerbiage(dispense)} -{' '}
                   {formatDatetime(parseDate(dispense.whenHandedOver))}
                 </h5>
-                <Tile className={styles.dispenseTile}>
+                <Tile className={styles.dispenseTile} data-floating-menu-container>
```
Comment (ibacher) [inline src/history/history-and-comments.component.tsx:235]: What's the data attribute for?

**Review by ibacher — CHANGES_REQUESTED** [review]
One small fix. Also, this PR needs to fix the existing conflicts before we can merge it.

File: src/history/history-and-comments.component.tsx (hunk 2 above)
Comment (donaldkibet) [reply src/history/history-and-comments.component.tsx:235]: `OverflowMenu` component uses react portal to render and needs a parent with the `data-floating-menu-container` attribute [docs](https://react.carbondesignsystem.com/?path=/docs/components-overflowmenu--overview#data-floating-menu-container)

**Review by ibacher — APPROVED** [review]

**Pull request merged**